src/code/audio/audio_seqplayer.c

```diff
@@ -1,73 +1,2310 @@
+    CMD_ARGS_1(u8),         // 0xE6 (channel: set book offset)
+    CMD_ARGS_1(s16),        // 0xE7 (channel:)
+    CMD_ARGS_3(u8, u8, u8), // 0xE8 (channel:)
+    CMD_ARGS_1(u8),         // 0xE9 (channel: set note priority)
+    CMD_ARGS_0(),           // 0xEA (channel: stop script)
+    CMD_ARGS_2(u8, u8),     // 0xEB (channel: set soundFont and instrument)
+    CMD_ARGS_0(),           // 0xEC (channel: reset vibrato)
+    CMD_ARGS_1(u8),         // 0xED (channel: set hilo gain)
+    CMD_ARGS_1(u8),         // 0xEE (channel: small bend pitch)
+    CMD_ARGS_2(s16, u8),    // 0xEF ()
+    CMD_ARGS_0(),           // 0xF0 (channel: unreserve notes)
+    CMD_ARGS_1(u8),         // 0xF1 (channel: reserve notes)
+    // Control flow instructions (>= 0xF2) can only have 0 or 1 args
+    CMD_ARGS_1(u8),  // 0xF2 (branch relative if less than zero)
+    CMD_ARGS_1(u8),  // 0xF3 (branch relative if equal to zero)
+    CMD_ARGS_1(u8),  // 0xF4 (jump relative)
+    CMD_ARGS_1(s16), // 0xF5 (branch if greater than or equal to zero)
+    CMD_ARGS_0(),    // 0xF6 (break)
+    CMD_ARGS_0(),    // 0xF7 (loop end)
+    CMD_ARGS_1(u8),  // 0xF8 (loop)
+    CMD_ARGS_1(s16), // 0xF9 (branch if less than zero)
+    CMD_ARGS_1(s16), // 0xFA (branch if equal to zero)
+    CMD_ARGS_1(s16), // 0xFB (jump)
+    CMD_ARGS_1(s16), // 0xFC (call and jump to a function)
+    CMD_ARGS_0(),    // 0xFD (delay n frames)
+    CMD_ARGS_0(),    // 0xFE (delay 1 frame)
+    CMD_ARGS_0(),    // 0xFF (end script)
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/audio_seqplayer/func_80197B14.s")
+/**
+ * Read and return the argument from the sequence script for a control flow instruction.
+ * Control flow instructions (>= 0xF2) can only have 0 or 1 args.
+ * @return the argument value for a control flow instruction, or 0 if there is no argument
+ */
+u16 AudioSeq_GetScriptControlFlowArgument(SeqScriptState* state, u8 cmd) {
+    u8 highBits = sSeqInstructionArgsTable[cmd - 0xA0];
+    u8 lowBits = highBits & 3;
+    u16 cmdArg = 0;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/audio_seqplayer/func_80197C0C.s")
+    // only 1 argument
+    if (lowBits == 1) {
+        if (!(highBits & 0x80)) {
+            cmdArg = AudioSeq_ScriptReadU8(state);
+        } else {
+            cmdArg = AudioSeq_ScriptReadS16(state);
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/audio_seqplayer/func_80197C8C.s")
+    return cmdArg;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/audio_seqplayer/AudioSeq_SequencePlayerDisableAsFinished.s")
+/**
+ * Read and execute the control flow sequence instructions
+ * @return number of frames until next instruction. -1 signals termination
+ */
+s32 AudioSeq_HandleScriptFlowControl(SequencePlayer* seqPlayer, SeqScriptState* state, s32 cmd, s32 cmdArg) {
+    u32 depth;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/audio_seqplayer/AudioSeq_SequencePlayerDisable.s")
+    switch (cmd) {
+        case 0xFF: // end script
```

Comment:
It is very much an open question as to what to do with these cmd hex values. The motivation of this PR in OoT is to think about what to do with these cmd values, and whether we want defines or just comments. Discussion can happen over on that PR, but I think I'd prefer to leave it as is until that discussion is resolved. 

https://github.com/zeldaret/oot/pull/1248

Reply:
We don't have to do the same thing as OOT. 

Everywhere else in the MM codebase we're using enums for switches where it matches and makes sense. I see no reason we wouldn't do the same here.

Reply:
Except that's my own OoT PR. The whole point of it being a draft is that enums might not even make sense due to the reuse op commands for this language, as there are three independent languages. Nearly all these commands are identical code, it'd be silly to have different names from OoT, especially in the audio library. And we can't name it from MM's perspective because we haven't started looking at audio extraction of sequences here.

Reply:
I wasn't suggesting that enums need to be done today, my apologies. My point is that eventually it should.